notify-api/src/notify_api/core/jwt.py

```diff
@@ -0,0 +1,208 @@
+"""Generic middleware."""
+import json
+import logging
+
+import jwt
+from jwt.algorithms import RSAAlgorithm
+from six.moves.urllib.request import urlopen
+from starlette.authentication import AuthCredentials, AuthenticationBackend, AuthenticationError, BaseUser
+from werkzeug.contrib.cache import SimpleCache
+
+from notify_api.core.settings import get_api_settings
+
+
+logger = logging.getLogger(__name__)
+
+
+class JWTUser(BaseUser):
+    """JWT User."""
+
+    def __init__(self, username: str, token: str, payload: dict) -> None:
+        """Init."""
+        self.username = username
+        self.token = token
+        self.payload = payload
+
+    @property
+    def is_authenticated(self) -> bool:
+        """Authenticated."""
+        return True
+
+    @property
+    def display_name(self) -> str:
+        """Display name."""
+        return self.username
+
+    @property
+    def identity(self) -> str:
+        """Identity."""
+        raise NotImplementedError()  # pragma: no cover
+
+
+class JWTWrapper:  # pylint: disable=too-few-public-methods
```

Comment:
Can we just extend thor's flask-oidc package here ?

Reply:
Thor's package is for flask app. I copied most of functions into here and using a public key to decode token during testing.